### ISSUE TITLE
Using an explicit --config-file on the commandline isn't handled properly

### ISSUE THREAD

**Issue by onlynone** [SuitestAutomation/suitest-js-api#449]
The `composeConfig` function: https://github.com/SuitestAutomation/suitest-js-api/blob/8b09d953d3cb05c85e8aebf541b816e3b1ce8979/lib/testLauncher/composeConfig.js#L231 has some confusing behavior.

It starts off like this:
```
const composeConfig = (argv) => {
	const rcConfig = readRcConfig(argv.configFile);
	const configFilePath = argv.configFile || rcConfig.configFile;
	const userConfig = configFilePath ? readUserConfig(configFilePath) : {};

	const ownArgs = {
		...pickNonNil(suitest.configuration.overridableFields, rcConfig),
		...pickNonNil(suitest.configuration.overridableFields, userConfig),
		...pickNonNil(suitest.configuration.overridableFields, argv),
	};
```

But if you pass `--config-file` into the program, then it looks like `const rcConfig = readRcConfig(argv.configFile);` will parse that file as any of the available config file formats. And then it looks like `configFilePath` will be set to the same path, and it'll be parsed again by the line `const userConfig = configFilePath ? readUserConfig(configFilePath) : {};`. Except this time, `readUserConfig` will only try to parse the file as json:

```
function readUserConfig(path) {
	try {
		return JSON.parse(fs.readFileSync(path));
	} catch (error) {
		throw new SuitestError(invalidUserConfig(path, error.message), error.code);
	}
}
```

So you'll get an exception if your config file is in json5 or yaml or something else. Here's an example run that shows this:

```
$ ./node_modules/.bin/suitest -v
3.1.5
$ cat suitest-config.json5
{
  // This is just a test
  "tokenId": "foo",
  "tokenPassword": "bar",
}
$ ./node_modules/.bin/suitest run --config-file suitest-config.json5 mocha
SuitestError: Failed to process config file 'suitest-config.json5'.
	Unexpected token / in JSON at position 4.
	Make sure path is correct and file is in valid json format.
...
```

I think if the config file has already been parsed by `readRcConfig`, then you don't want to try parsing it again and `readUserConfig` should probably be written to accept all the different formats that `readConfigFile` accepts, no?

**Comment by onlynone** [comment]
I just reproduced this error with 3.6.1. Here's my `package.json`

**Comment by onlynone** [comment]
I just reproduced this with 3.6.1. Here's my `package.json`:
```json
{
  "name": "e2e-suitest3",
  "version": "1.0.0",
  "main": "index.js",
  "license": "MIT",
  "dependencies": {
    "suitest-js-api": "^3.6.1"
  }
}
```
And here's my `suitest-config.json5`:
```json5
{
  // This is just a test
  "tokenId": "foo",
  "tokenPassword": "bar",
}
```
And here's trying to run with `--config-file`:
```
$ yarn run suitest run --config-file suitest-config.json5
yarn run v1.22.17
warning ../../package.json: No license field
$ /Users/steven/projects/e2e-suitest3/node_modules/.bin/suitest run --config-file suitest-config.json5
suitest run

Command for start testing

Options:
  --help, -h                Show help                                  [boolean]
  --version, -v             Show version number                        [boolean]
  --token-id, -k            Suitest Token Key, can be generated on the profile
                            page
  --token-password, -p      Suitest Token Password, can be generated on the
                            profile page
  --device-id, -d           Device you want to connect to
  --app-config-id, -c       Application configuration id to launch the app with
  --preset                  Preset for configuration and device.         [array]
  --inspect                 Will launch user command with --inspect execArgv,
                            used for debugging
  --inspect-brk             Will launch user command with --inspect-brk
                            execArgv, used for debugging
  --config-file             Config file to override default config.     [string]
  --default-timeout         Globally set timeout value in milliseconds. Equals
                            to 2000 when left out.                      [number]
  --disallow-crash-reports  Disallow Suitest to submit automatic crash reports
                            when unexpected exceptions occur           [boolean]
  --log-dir                 If defined, additional logging to specified folder
                            will be performed.
  --log-level               Logger verbosity level [default: "normal"]
                      [choices: "silent", "normal", "verbose", "debug", "silly"]
  --timestamp               Logger timestamp format. Use "none" to disable.
                            [default: "MMM D HH:mm:ss Z"]. This string is
                            processed with momentJS.                    [string]
  --screenshot-dir          Directory for saving screenshots.           [string]
  --concurrency             How many parallel sessions of the test runner should
                            be spawned. I.e. on how many devices will it run in
                            parallel. 0 = unlimited                 [default: 0]
  --include-changelist      If unapplied changes should be tested with
                            runSnippet command.       [boolean] [default: false]

SuitestError: Failed to process config file 'suitest-config.json5'.
	Unexpected token / in JSON at position 4.
	Make sure path is correct and file is in valid json format.
    at Object.<anonymous> (/Users/steven/projects/e2e-suitest3/node_modules/ramda/src/once.js:40:17)
    at Object.handler (/Users/steven/projects/e2e-suitest3/node_modules/ramda/src/internal/_arity.js:11:19)
    at Object.runCommand (/Users/steven/projects/e2e-suitest3/node_modules/yargs/build/lib/command.js:196:48)
    at Object.parseArgs [as _parseArgs] (/Users/steven/projects/e2e-suitest3/node_modules/yargs/build/lib/yargs.js:1043:55)
    at Object.parse (/Users/steven/projects/e2e-suitest3/node_modules/yargs/build/lib/yargs.js:584:29)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/steven/projects/e2e-suitest3/node_modules/suitest-js-api/bin/suitest:3:1)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Module.load (internal/modules/cjs/loader.js:928:32)
    at Function.Module._load (internal/modules/cjs/loader.js:769:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:72:12)
    at internal/main/run_main_module.js:17:47 {
  code: Symbol(unknownError),
  type: 'SuitestError'
}
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by polomsky** [comment]
Currently docs says about `--config-file` option: _Path to JSON config file to override the [configuration file](https://suite.st/docs/suitest-api/setup/#the-configuration-file) config_

So marking this as _enhancement_.

**Comment by polomsky** [comment]
@onlynone The code really does not make any sense to me. What is the usecase for using `--config-file` ?
In 3.x version, no other config file is read (as mentioned in [docs](https://suite.st/docs/suitest-api/advanced-topics/#locations-for-configuration-files).
Do you want to force a single configuration file and ignore the rest or you want to override standard configuration file with your settings as mentioned in [launcher options](https://suite.st/docs/suitest-api/test-launcher/#launcher-options) and that is how 2.x version works?

**Comment by onlynone** [comment]
The documentation you link to also says that the file can be all sorts of different formats. It looks like maybe it's saying you'd have to use the right file extension. But I think, based on what I remember of the code, it doesn't matter what the file extension is.

> @onlynone The code really does not make any sense to me. What is the usecase for using `--config-file` ?

I was exploring it as a way to use alternate config values in different contexts. Like if it's running under our CI system, maybe we'd run tests with `--config-file .ci-suitestrc` and our CI system would be configured to make sure that that file existed with appropriate values prior to the run. When developers are running the tests from their own computer, maybe we'd use `--config-file .dev-suitestrc` and that file would be written to have base settings ideal for interactive developer use and maybe load `~/.suitestrc` in order to get the user's username and password (or something like that).

With v2, we could do all that by just setting certain environment variables and creating `.suitestrc` files in the right places and all the config values would be automatically merged.

> In 3.x version, no other config file is read (as mentioned in [docs](https://suite.st/docs/suitest-api/advanced-topics/#locations-for-configuration-files). Do you want to force a single configuration file and ignore the rest or you want to override standard configuration file with your settings as mentioned in [launcher options](https://suite.st/docs/suitest-api/test-launcher/#launcher-options) and that is how 2.x version works?

I'm just trying to figure out the simplest way to use different values for the config in different contexts in the simplest way possible. I saw the `--config-file` option and thought it might provide a way to do that.

**Comment by polomsky** [comment]
All issues are addressed by release [3.7.0](https://github.com/SuitestAutomation/suitest-js-api/releases/tag/v3.7.0).
There are two new CLI parameters `--base-config-file` and `override-config-file` which has specific behaviour. More details can be found in [launcher options](https://suite.st/docs/suitest-api/test-launcher/#launcher-options) and in [advance topics](https://suite.st/docs/suitest-api/advanced-topics/#locations-for-configuration-files)

Also both parameters now support all variation of configuration files.
